docs/resources/server/advancements.md

```diff
@@ -158,35 +158,60 @@ Advancements can be [datagenned][datagen] using an `AdvancementProvider`. An `Ad
 :::
 
-To start, create a subclass of `AdvancementProvider`:
+To start, create an instance of `AdvancementProvider` within `GatherDataEvent`:
 
 ```java
-public class MyAdvancementProvider extends AdvancementProvider {
-    // Parameters can be obtained from GatherDataEvent.
-    public MyAdvancementProvider(PackOutput output,
-            CompletableFuture<HolderLookup.Provider> lookupProvider, ExistingFileHelper existingFileHelper) {
-        super(output, lookupProvider, existingFileHelper, List.of());
-    }
+@SubscribeEvent
+public static void gatherData(GatherDataEvent event) {
+    DataGenerator generator = event.getGenerator();
+    PackOutput output = generator.getPackOutput();
+    CompletableFuture<HolderLookup.Provider> lookupProvider = event.getLookupProvider();
+    ExistingFileHelper existingFileHelper = event.getExistingFileHelper();
+
+    generator.addProvider(
+            event.includeServer(),
+            new AdvancementProvider(
+                output, lookupProvider, existingFileHelper,
+                // Add generators here
+                List.of(...)
+            )
+    );
+
+     // Other providers
 }
 ```
 
-Now, the next step is to fill the list with our generators. To do so, we add one or more generators as static classes and then add an instance of each of them to the currently empty list in the constructor parameter.
+Now, the next step is to fill the list with our generators. To do so, we can either add generators as classes or lambdas, and then add an instance of each of them to the currently empty list in the constructor parameter.
 
 ```java
-public class MyAdvancementProvider extends AdvancementProvider {
-    public MyAdvancementProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> lookupProvider, ExistingFileHelper existingFileHelper) {
-        // Add an instance of our generator to the list parameter. This can be done as many times as you want.
-        // Having multiple generators is purely for organization, all functionality can be achieved with a single generator.
-        super(output, lookupProvider, existingFileHelper, List.of(new MyAdvancementGenerator()));
-    }
+// Class example
+public class MyAdvancementGenerator extends AdvancementProvider.AdvancementGenerator {
 
-    private static final class MyAdvancementGenerator implements AdvancementProvider.AdvancementGenerator {
-        @Override
-        public void generate(HolderLookup.Provider registries, Consumer<AdvancementHolder> saver, ExistingFileHelper existingFileHelper) {
-            // Generate your advancements here.
-        }
+    @Override
+    public void generate(HolderLookup.Provider registries, Consumer<AdvancementHolder> saver, ExistingFileHelper existingFileHelper) {
+        // Generate your advancements here.
     }
 }
+
+// Method Example, assume in ExampleClass
```

Comment:
Actually do a full `public class` declaration here, for making it easier to read.